### PR TITLE
Product image getting removed in the new product editor after replacing cover image

### LINKED ISSUE #49155: Product image getting removed in the new product editor after replacing cover image
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When in the new product editor and replacing cover image, it happens that the previously set image cover disappear after replacing the current one.

### Expected behavior

No images removed/disappear

### Actual behavior

Image disappear/removed

### Steps to reproduce

1. Go to the new product editor
2. Make sure to have some images in the media library
3. Add 4-5 product images
4. Set some as cover
5. Click cover image and replace it with any other image from the library
6. Observe issue

https://d.pr/i/Cvo3A1

It might be that you cant reproduce it at first, but try a few times.

### WordPress Environment

clean install Jurassic site

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #49155

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the new product editor
2. Make sure to have some images in the media library
3. Add 4-5 product images
4. Set some as cover
5. Click cover image and replace it with any other image from the library
6. Check that moving any image to a different position is working correctly
8. Check that setting any image as cover is working correctly
9. Check that deleting any image is working correctly
10. Check that any combination of the previous features is working correctly

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by louwie17 — APPROVED** [review]
Approving this as it appears to address the bugs shown as part of the original issue. Although I did notice another issue while testing this, if you have multiple images and replace an image with any of the other images ( or the same one ) it replaces all the images and only keeps the individual image ( see video ):

https://github.com/user-attachments/assets/a1f40a46-66fb-4b04-a31c-4008aa6aa9d3

Also quick side note, from the code changes it was really hard to tell what actually addressed the issue ( I am guessing removing the internal state of `orderedChildren` and relying on the passed in `children` property is what fixed it. But, it would be helpful to provide some additional context as part of GH comments within the code.

**Comment by mdperez86** [comment]
> Approving this as it appears to address the bugs shown as part of the original issue. Although I did notice another issue while testing this, if you have multiple images and replace an image with any of the other images ( or the same one ) it replaces all the images and only keeps the individual image ( see video ):
> 
>  image-replace-issue.mp4 
> Also quick side note, from the code changes it was really hard to tell what actually addressed the issue ( I am guessing removing the internal state of `orderedChildren` and relying on the passed in `children` property is what fixed it. But, it would be helpful to provide some additional context as part of GH comments within the code.

Nice catch @louwie17 

Done -> [c1932e8](https://github.com/woocommerce/woocommerce/pull/50181/commits/c1932e8c9c8bf0d2f76c8bcb57a9bf26f3db612b)

File: packages/js/product-editor/src/blocks/product-fields/images/edit.tsx [hunk 1]
```diff
@@ -162,10 +162,12 @@ export function ImageBlockEdit( {
...
 		recordEvent( 'product_images_replace_image_button_click' );
 
-		if (
-			Array.isArray( propertyValue ) &&
-			! propertyValue.some( ( img ) => media.id === img.id )
-		) {
+		if ( Array.isArray( propertyValue ) ) {
+			// Ignore the media if it is replaced by itseft.
+			if ( propertyValue.some( ( img ) => media.id === img.id ) ) {
+				return;
```
Comment (louwie17) [inline packages/js/product-editor/src/blocks/product-fields/images/edit.tsx:168]: Fixes the issue, nice!
Ideally we would show a notice letting the user know that the image was already added.

**Review by louwie17 — APPROVED** [review]
Re-approving, thanks for fixing the above bug, it tested well now.
I did leave one final comment 🙈

**Pull request merged**